To-do-list/Controller/TaskViewController.swift

```diff
@@ -45,12 +42,9 @@ class TaskViewController: UIViewController, UITextFieldDelegate, UITextViewDeleg
     @IBAction func saveButtonTapped(_ sender: Any) {
         guard let title = titleField.text, !title.isEmpty else { return }
         guard let description = descriptionField.text, !description.isEmpty else { return }
-        
-        let taskDictionnary = ["title": title, "description": description]
-        UserDefaults.standard.set(taskDictionnary, forKey: "newTask")
-        
-        completionHandler?()
-        navigationController?.popToRootViewController(animated: true)
+        taskViewModel?.saveTask(title: title, description: description)
+        taskViewModel?.setupTextFieldAppearence(titleField: titleField, descriptionField: descriptionField)
+        taskViewModel?.returnToMainViewController()
```

Comment:
think, that saveTask and setupTextFieldAppearance -  it unnecessary to cal from viewController

Reply:
Why? Like this I set the input text in array. And the text field appearance needed to be corrected as soon as I started to do navigation via AppDelegate